Firebase V3 SDK for the cyclic-fire driver

jemmyw opened the pull request:
Small changes to the cyclic-fire driver made locally so that it works iwth the latest Firebase SDK.

jemmyw committed:
feat(Firebase): Update cyclic to work with firebase 3 and implement google login button

jemmyw committed:
fix(FirebaseAuth): use global auth change event so that we know if you are already logged in

jemmyw committed:
chore(FirebaseQueue): Update cyclic queue to work with firebase 3

TylorS commented on src/cyclic-fire/index.js:
```diff
@@ -0,0 +1,132 @@
+
+const ValueStream = ref => FirebaseStream(ref, 'value')
+  .map(({val}) => val)
+  .memory()
```
I think this should be `.remember()`

jemmyw committed:
fix(Firebase): memory should be remember

jemmyw committed:
feat(Firebase): Add documentation for running to the readme

TylorS committed:
chore(package): update @cycle/dom dependency

* chore(package): update @cycle/dom dependency

This has a fix for our bug regarding logout

* fix(Landing): use new firebase v3 email location

user.email will return null, use user.providerData[0].email to get proper access to email address
used to login with.

* test(e2e): add e2e tests for the firebase login process

Adds a common function that can be used to automate loggin in as a user. Requires for
TEST_ACCOUNT_EMAIL and TEST_ACCOUNT_PASSWORD be defined in ENV variables.

* refactor(Landing): remove console.log
